rules/providers.bzl

```diff
@@ -1,6 +1,11 @@
 FrameworkInfo = provider(
     fields = {
         "vfsoverlay_infos": "Merged VFS overlay infos, present when virtual frameworks enabled",
-        "framework_headers": "Headers part of the framework's public interface",
+        "binary": "The binary of the framework",
+        "headers": "Headers of the framework's public interface",
+        "private_headers": "Private headers of the framework",
+        "modulemap": "The module map of the framework",
+        "swiftmodule": "The swiftmodule",
+        "swiftdoc": " The Swift doc",
```

Comment:
what;s the relationship between swiftmoduel and swiftdoc? right now based on what i have seen it feels like swiftdoc is a child of swiftmodule... along with swiftinterface

Reply:
I guess that's just how they're packaged in a framework.

Reply:
They're all binary formats, you can read more about swift modules here: https://github.com/apple/swift/blob/main/docs/Serialization.md

(TL;DR functionally they're like headers that expose public APIs)

Also IIRC .swiftdoc files hold documentation related things like when you use "///" to document a public class and want its description to show up in the UI for who is consuming it (it could be holding more info than that but AFAIK this format is not documented)